Convert metasearch button popup to a modal

chenikabukes opened the pull request:
<!-- prettier-ignore -->
## Fixes
<!-- If PR doesn't fully resolve the issue, replace 'Fixes' below with 'Related to'. -->
<!-- If there is no issue being resolved, please consider opening one before creating this pull request. -->

Fixes #2668 by @fcoveram 

## Description
<!-- Concisely describe what the pull request does. -->
<!-- Add screenshots, videos, or other media to show the problem and the solution when appropriate. -->
This PR addressed the design improvement in #460 by making the external sources popup a modal and adjusting the modal title according to screen size.

## Testing Instructions
<!-- Give steps for the reviewer to verify that this PR fixes the problem; or delete this section entirely. -->
1. Conduct a search
2. Scroll to the bottom to find the new external search button
3. Click on the button to let the modal display
4. Change screen size to view the different modal titles according to screen size

## Checklist
<!-- Replace  the [ ] with [x] to check the boxes. -->

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [x] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin
<!-- You must read and understand the following attestation. -->

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

obulat commented on frontend/src/components/VExternalSearch/VExternalSearchForm.vue:
```diff
@@ -5,62 +5,70 @@
     <template v-if="triggerElement">
       <Component
-        :is="isMd ? 'VPopoverContent' : 'VModalContent'"
-        :id="isMd ? 'external-sources-popover' : 'external-sources-modal'"
+        :is="'VModalContent'"
```
Since this is only a `VModalContent` component now, we don't need to use `<Component is='VModalContent'` anymore. We can use `<VModalContent` here.

obulat commented on frontend/src/components/VExternalSearch/VExternalSearchForm.vue:
```diff
@@ -5,62 +5,70 @@
-        :is="isMd ? 'VPopoverContent' : 'VModalContent'"
-        :id="isMd ? 'external-sources-popover' : 'external-sources-modal'"
+        :is="'VModalContent'"
+        :id="'external-sources-modal'"
```
```suggestion
        id="external-sources-modal"
```

obulat commented on frontend/src/components/VExternalSearch/VExternalSearchForm.vue:
```diff
@@ -5,62 +5,70 @@
         :visible="isVisible"
-        :z-index="isMd ? 'popover' : 'modal'"
-        :variant="isMd ? undefined : 'centered'"
+        :z-index="'modal'"
```
```suggestion
        z-index="modal"
```

obulat commented on frontend/src/components/VExternalSearch/VExternalSearchForm.vue:
```diff
@@ -5,62 +5,70 @@
-        :z-index="isMd ? 'popover' : 'modal'"
-        :variant="isMd ? undefined : 'centered'"
+        :z-index="'modal'"
+        :variant="'centered'"
```
```suggestion
        variant="centered"
```
Static values that do not change based on a variable value can be simplified like this

obulat reviewed (changes requested):
This is a really great start, @chenikabukes! I've added a couple of change requests inline mainly for cleaning up the dynamic values that are no longer necessary. I will do a more thorough review tomorrow.

fcoveram commented:
Why was this ticket closed?

chenikabukes commented:
> Why was this ticket closed?

I just closed the PR, not the ticket, rebased onto main and addressed the comments in a new in progress PR: https://github.com/WordPress/openverse/pull/2811#issuecomment-1681413931

fcoveram commented:
I see 👍  Thanks @chenikabukes
